### PR TITLE
ci: upload artifacts for emerynet tests to GCS

### LINKED ISSUE #98: Upload artifacts of emerynet e2e test to GCP
**Is your feature request related to a problem? Please describe.**
#97 introduces emerynet tests for the repo, but due to the fact that we are using mnemonic phrases that need to be kept private, the screen recording artifacts cannot be made public. therefore we need to upload them to a server that only Agoric OpCo can access.

**Describe the solution you'd like**
The solution is to upload them to Google cloud storage using the https://github.com/google-github-actions/upload-cloud-storage github action.

### PR DESCRIPTION
**Pull request by frazarshad**
Fixes #98 

Allows us to upload the video artifacts for emerynet tests to a private GCS bucket

### CODE REVIEW

**Comment by samsiegart** [comment]
Adding @toliaqat because I'm not too familiar with GCS. I'd imagine we want some documentation to explain how to access these artifacts there, not sure if it makes sense to put that documentation in the repo publicly.

File: .github/workflows/e2e_tests.yml [hunk 1]
```diff
@@ -70,3 +81,11 @@ jobs:
             tests/e2e/docker/videos
             tests/e2e/docker/screenshots
         continue-on-error: true
+
+      - name: Archive e2e artifacts to GCS
+        uses: google-github-actions/upload-cloud-storage@v2
+        if: github.event_name == 'schedule'
+        with:
+          path: 'tests/e2e'
+          destination: 'github-artifacts/${{ github.repository }}/${{ github.run_id }}/${{ github.event.repository.updated_at }}'
```
Comment (toliaqat) [inline .github/workflows/e2e_tests.yml:90]: We may want to add a log entry here that prints the upload path which someone with permissions can click to open and watch the recordings.

**Review by toliaqat — COMMENTED** [review]
Left a comment print the path of the upload location.

File: .github/workflows/e2e_tests.yml [hunk 2]
```diff
@@ -17,6 +17,10 @@ jobs:
   e2e:
     runs-on: ubuntu-latest
 
+    permissions:
+      contents: 'read'
+      id-token: 'write'
```
Comment (toliaqat) [inline .github/workflows/e2e_tests.yml:22]: Why do we need `id-token: 'write'` permission?

**Comment by toliaqat** [comment]
> Adding @toliaqat because I'm not too familiar with GCS. I'd imagine we want some documentation to explain how to access these artifacts there, not sure if it makes sense to put that documentation in the repo publicly.

Asked in another comment to print the path to GCS in the action logs for now.

File: .github/workflows/e2e_tests.yml (hunk 1 above)
Comment (frazarshad) [reply .github/workflows/e2e_tests.yml:90]: it already the paths of all uploaded files in the logs: 
image

Comment (toliaqat) [reply .github/workflows/e2e_tests.yml:90]: Is there a possibility to print a HTTP path to the video folder of this job?
Example:
https://console.cloud.google.com/storage/browser/github-artifacts/Agoric/dapp-econ-gov/8817132062/2024-04-24T12:12:59Z/e2e/docker/videos

Comment (toliaqat) [reply .github/workflows/e2e_tests.yml:90]: This UX consideration, when Brendan comes to this job to check the videos, they might not know the full path and gs:// might be strange to them.

File: .github/workflows/e2e_tests.yml (hunk 2 above)
Comment (frazarshad) [reply .github/workflows/e2e_tests.yml:22]: this is the default permissions required by upload-cloud-storage action as shown in [this example](https://github.com/google-github-actions/upload-cloud-storage?tab=readme-ov-file#authorization)
searching further, [this document](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#adding-permissions-settings) details that this permission is required in order to perform JWT authentication

File: .github/workflows/e2e_tests.yml (hunk 1 above)
Comment (frazarshad) [reply .github/workflows/e2e_tests.yml:93]: updated
image

**Review by toliaqat — APPROVED** [review]

**Pull request merged**